### PR TITLE
Refreshes main README about Request tracing

### PR DESCRIPTION
**Pull request by adriancole**
This attempts to clarify the main README to reduce grief for folks who read it.
Particularly, sometimes we get people modeling different things, such as local
activity or long-lived connections as CLIENT spans.

See https://github.com/openzipkin/zipkin-support/issues/2

### CODE REVIEW

File: brave/README.md [hunk 1]
```diff
@@ -167,37 +167,59 @@ void userCode() {
...
-// when the response is complete, finish the span
-span.finish();
-```
+You should use these abstractions instead of modeling your own, as there are
+modeling gotchas that might not be intuitive at first. For example, a common
+mistake is using CLIENT kind for long lived connections such as database pools
+or chat sessions. Doing so, however, can result in confusing data, such as
+incorrect service diagrams or server calls being attached to the wrong trace.
+If you have doubts, please chat with a maintainer on [Gitter](https://gitter.im/openzipkin/zipkin]
+as it might save you grief later!
```
Comment (jorgheymans) [inline brave/README.md:222]: i would move this block above 'once you model your request'

File: brave/README.md [hunk 2]
```diff
@@ -167,37 +167,59 @@ void userCode() {
...
+communication `brave.Request` and `brave.Response`. These types represent the
+following communication patterns, defined by the [Zipkin Api][https://zipkin.io/zipkin-api/#/default/post_spans]
+and returned by `spanKind()`:
+
+ * CLIENT
+ * SERVER
+ * PRODUCER
+ * CONSUMER
+
+Once you model your request, you generally need to...
```
Comment (jorgheymans) [inline brave/README.md:184]: "If you really think that the existing abstractions do not match your need, and you want to model your request yourself, you generally ...."

File: brave/README.md [hunk 3]
```diff
@@ -167,37 +167,59 @@ void userCode() {
...
+  span.finish(); // 5.
+}
+```
 
-// if there is an error, tag the span
-span.tag("error", error.getCode());
+The above code is example only. Abstractions exist that handle aspects
+including sampling, header processing and data parsing:
+ * [HTTP](../instrumentation/http/README.md) is CLIENT/SERVER
+ * [Messaging](../instrumentation/rpc/README.md) is PRODUCER/CONSUMER
```
Comment (anuraaga) [inline brave/README.md:213]: Wrong href

Comment (adriancole) [reply brave/README.md]: thx fixed in latest.  I realized we repeated a lot of this in the instrumentation README!

File: instrumentation/README.md [hunk 4]
```diff
@@ -95,14 +95,52 @@ We worked very hard to make writing new instrumentation easy and efficient.
...
+
+Using abstractions also helps avoid modeling gotchas that might not be
+intuitive at first.
+
+For example, a common mistake is using CLIENT kind for long lived connections
+such as database pools or chat sessions. Doing so, however, can result in
+confusing data, such as incorrect service diagrams or server calls being
+attached to the wrong trace.
+
+It is also a common misconception that a grouping of CLIENT spans should itself
```
Comment (adriancole) [inline instrumentation/README.md:127]: @anuraaga this bud's for you

File: instrumentation/README.md [hunk 5]
```diff
@@ -95,14 +95,52 @@ We worked very hard to make writing new instrumentation easy and efficient.
...
+to look for tests in the "features" package before deciding something doesn't
+work.
+* [Core feature tests](../brave/src/test/java/brave/features)
+* [HTTP feature tests](http-tests/src/test/java/brave/http/features)
+* [Messaging feature tests](messaging/src/test/java/brave/rpc/features)
+* [RPC feature tests](rpc/src/test/java/brave/rpc/features)
+
+If you end up in a position where you still believe you need a custom model,
+please contact us before committing to it. There may be an alternative not yet
+documented, that might prefer. Moreover, folks usually want support and we
```
Comment (adriancole) [inline instrumentation/README.md:143]:
```suggestion
documented, you might prefer. Moreover, folks usually want support and we
```

**Comment by adriancole** [comment]
thanks for the feedback @jorgheymans and @anuraaga!

**Pull request merged**